### PR TITLE
Documentation using javadoc

### PR DESCRIPTION
**Pull request by fab-10**

### CODE REVIEW

File: sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java [hunk 1]
```diff
@@ -62,6 +71,19 @@ public ProfitableTransactionSelector(
...
+   * retried, since the gas price market could now make it profitable, but only a configurable
+   * amount of these transactions is retried each time, to avoid that they could potentially consume
+   * all the time allocated to block creation.
+   *
+   * @param evaluationContext The current selection context.
+   * @return TX_UNPROFITABLE_UPFRONT if the transaction is not profitable upfront,
+   *     TX_UNPROFITABLE_RETRY_LIMIT if the transaction was already found to be unprofitable, and
+   *     there are no more slot to retry past unprofitable transactions during this block creation
+   *     process, otherwise SELECTED.
+   */
```
Comment (ahamlat) [inline sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java:86]: From the implementation below, I understand that unprofitableRetries is for all transactions. By default we can retry 10 times one specific or different transactions. This counter is initialized to 0 for each block creation while the cache remain static. Is my understanding correct ?

File: sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java [hunk 2]
```diff
@@ -27,13 +27,29 @@
...
 @Slf4j
 @RequiredArgsConstructor
 public class MaxBlockGasTransactionSelector implements PluginTransactionSelector {
 
   private final long maxGasPerBlock;
   private long cumulativeBlockGasUsed;
 
+  /**
+   * Evaluates a transaction post-processing. Checks if adding the gas used of the transaction, to
+   * the cumulative gas used of the block till now, is below the user configured max gas used per
```
Comment (ahamlat) [inline sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java:45]: Does user here mean, the node that is going to create the block, basically the sequencer in Linea ?

**Review by ahamlat — APPROVED** [review]
few comments and questions

File: sequencer/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java [hunk 3]
```diff
@@ -23,6 +23,17 @@
 import org.hyperledger.besu.datatypes.Wei;
 import org.slf4j.spi.LoggingEventBuilder;
 
+/**
+ * This class implements the profitability formula, and it is used both to check if a tx is
+ * profitable and to give an estimation the profitable priorityFeePerGas for a given tx. The
```
Comment (fab-10) [inline sequencer/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java:28]:
```suggestion
 * profitable and to give an estimation of the profitable priorityFeePerGas for a given tx. The
```

File: sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java (hunk 2 above)
Comment (fab-10) [reply sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java:45]: it means who is running the sequencer, I was not sure about _user_, do you think _operator_ is better in this context?

File: sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java (hunk 1 above)
Comment (fab-10) [reply sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java:86]: Yes, a new instance of the selector is created on every new block creation process, so the cache is static to persist across block creations, while the retry counter is reset on every block creation, so we can retry at max 10 unprofitable tx per block creation.

**Pull request merged**